File: 아이지에이/1.py
t=int(input())
for _ in range(t):
    n,m=map(int,input().split())
    arr=list(map(int,input().split()))
    arr2=arr[::]
    check=[0]*m
    answer=0
    while True:
        answer+=1
        for i in range(m):
            if check[i]==0 and n>0:
                check[i]=1
                n-=1
            if check[i]==1:
                arr2[i]=max(0,arr2[i]-1)
            if arr2[i]==0:
                arr2[i]=arr[i]
                check[i]=0
        if n==0 and 1 not in check:
            break
    print(answer)

        

File: scpc2021/1.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int t,n;
    cin>>t;
    for(int i=0; i<t; i++){
        vector<int> arr;
        int answer=0;
        cin>>n;
        vector<int> check(n);
        for (int j=0;j<n;j++){
            int temp;
            cin>>temp;
            arr.push_back(temp);
        }
        
        for (int j=0;j<n;j++){
            if (check[j]==0){
                answer+=1;
                int x=j;
                while (true){
                    // cout<<j<<x<<endl;
                    if (check[x]==1){
                        answer-=1;
                        break;
                    }
                    check[x]=1;
                    x=x+arr[x];
                    if (x>=n){
                        break;
                    }    
                }
            }
        }
        cout << "Case #" << i+1 << endl;
		cout << answer << endl;
    }
}